### PR TITLE
Add API and TUI support for LUKS recovery key

### PR DESCRIPTION
**Pull request by ogayot**
This is basically working and ready to be reviewed IMO. Some more automated / units tests would be beneficial.

### CODE REVIEW

File: subiquity/models/filesystem.py [hunk 1]
```diff
@@ -47,6 +48,84 @@ class NotFinalPartitionError(Exception):
...
+    live_location: Optional[pathlib.Path]
+    # Where to store the key in the target system. /target will automatically
+    # be prefixed.
+    backup_location: pathlib.Path
+
+    _key: Optional[str] = attr.ib(repr=False, default=None)
+
+    @classmethod
+    def from_spec(
+        cls, spec: Optional[Dict[str, Any]]
```
Comment (dbungert) [inline subiquity/models/filesystem.py:63]: We do use these "spec" dictionaries for filesystem stuff, but want to move away from them.  Would you create a new attrs object containing your desired fields?

File: subiquity/models/subiquity.py [hunk 2]
```diff
@@ -176,11 +176,13 @@ def __init__(self, root, hub, install_model_names, postinstall_model_names):
             self.target = root
             self.chroot_prefix = []
 
+        log.debug("root is %s", root)
```
Comment (dbungert) [inline subiquity/models/subiquity.py:179]: debug stuff, is this meant for inclusion in the PR?

File: subiquity/ui/views/filesystem/lvm.py [hunk 3]
```diff
@@ -205,6 +221,17 @@ def done(self, sender):
...
         mdc = self.form.devices.widget
         result["devices"] = mdc.active_devices
+        if "create_recovery_key" in result:
+            if result["create_recovery_key"]:
+                backup_prefix = pathlib.Path("/var/log/installer")
+                backup_filename = pathlib.Path(f"recovery-key-{result['name']}.txt")
+                result["recovery-key"] = {
+                    "backup-location": str(backup_prefix / backup_filename),
+                    "live-location": str(
+                        pathlib.Path("~").expanduser() / backup_filename
```
Comment (dbungert) [inline subiquity/ui/views/filesystem/lvm.py:231]: the use of the term "backup" sounds like a copy-paste error but really it's fine, so maybe rename backup_filename to a term that users neither the word "backup" nor "live".

File: subiquity/server/controllers/install.py [hunk 4]
```diff
@@ -688,6 +688,7 @@ async def postinstall(self, *, context):
                     hostname = f.read().strip()
 
             await self.app.controllers.Ad.join_domain(hostname, context)
+        self.model.filesystem.copy_artifacts_to_target()
```
Comment (dbungert) [inline subiquity/server/controllers/install.py:691]: user hasn't been created yet

Comment (mwhudson) [reply subiquity/server/controllers/install.py:691]: but the key is getting copied to /var/log/installer right?

**Review by mwhudson — COMMENTED** [review]
I agree with Dan :-)

File: subiquity/models/subiquity.py (hunk 2 above)
Comment (ogayot) [reply subiquity/models/subiquity.py:179]: Nope! thanks for noticing!

File: subiquity/server/controllers/install.py (hunk 4 above)
Comment (ogayot) [reply subiquity/server/controllers/install.py:691]: Yes, it's exposed in $HOME in the live session, but then ends up copied to /var/log/installer in the target.

File: subiquity/ui/views/filesystem/lvm.py (hunk 3 above)
Comment (ogayot) [reply subiquity/ui/views/filesystem/lvm.py:231]: Good point! Renaming to "filename".

**Review by dbungert — APPROVED** [review]

**Pull request merged**